src/strings/costring.rs

```diff
@@ -0,0 +1,136 @@
+/// A null-terminated wide-string that has been allocated with `CoTaskMemAlloc` or `CoTaskMemRealloc`
+/// and is freed with `CoTaskMemFree`.
+#[repr(transparent)]
+pub struct CoString(*mut u16);
+
+impl CoString {
+    /// Create a new `CoString`
+    pub fn new() -> Self {
+        Self(std::ptr::null_mut())
+    }
+
+    // Create a CoString from an iterator of u16s and a length.
+    // The iterator must have as less than or equal number of elements as `len`
+    unsafe fn from_wide_iter<I>(iter: I, len: usize) -> Self
+    where
+        I: IntoIterator<Item = u16>,
+    {
+        let start = CoTaskMemAlloc(len * 2 + 2) as *mut u16;
+        assert!(!start.is_null(), "Could not allocate memory for CoString");
+        let mut cursor = start;
+        for (index, c) in iter.into_iter().enumerate() {
+            debug_assert!(index < len);
+            cursor.write(c);
+            cursor = cursor.add(1);
+        }
+        cursor.write(0);
+        Self(start)
+    }
+
+    /// Get the string as 16-bit wide characters (wchars).
```

Comment:
Perhaps just a comment explaining why this doesn't return a slice (because the size is not computed) as that would otherwise be very handy for Windows development. Fortunately HString provides this so there's no compelling need for CoString to do so.